guides/hack/56-modules/05-traits.md

```diff
@@ -0,0 +1,44 @@
+Traits have special behavior in Hack when used with modules. 
+
+## Marking traits internal 
+An internal trait can only be used within the module it's defined. Internal traits can have public, protected, or internal methods. 
+
+```hack
+module foo;
+internal trait TFoo {
+    public function foo(): void {}
+    internal function bar(): void {}
+}
+```
+
+```hack
+module bar;
+public class Bar {
+    use TFoo; // error, TFoo is internal
+}
+```
+
+
+
+## Public traits
+Since public traits have "copy-paste" behavior in that their implementations can be copied to any other module, these traits must be treated as if they are outside of any module for safety. Therefore, public traits cannot access any internal symbols or implementations from the module they are defined in. 
+
+```hack
+module foo;
+internal class Foo {}
+// Assume TFoo is defined as before
+public trait TBar {
+    use TFoo; // error, TBar is a public trait, it cannot access internal symbols in module foo
+    public function foo(): mixed {
+        return new Foo(); // error, TBar is a public trait, it cannot access internal symbols in module foo
+    }
+}
+```
+Public traits also cannot have internal members, as they may be used by classes outside of any module. 
```

Comment:
We don't generally use the term members for users, prefer 'methods or properties'.